packages/expo-sqlite/src/queryUtils.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * Information about a parsed SQL query
+ */
+export interface SQLParsedInfo {
+  canReturnRows: boolean;
+}
+
+const SINGLE_QUOTED_STRING = /'(?:[^']|'')*'/g;
+const DOUBLE_QUOTED_STRING = /"(?:[^"]|"")*"/g;
+const RETURNING_KEYWORD = /\bRETURNING\b/i;
+const MUTATION_KEYWORDS = /\b(INSERT|UPDATE|DELETE|CREATE|ALTER|DROP)\b/i;
+const QUERY_KEYWORDS = /\b(SELECT|PRAGMA|WITH|EXPLAIN)\b/i;
+
+/**
+ * Parse SQL query to determine if it can return rows
+ * Uses a priority-based approach for accurate detection
+ *
+ * A reimplementation of Bun's SQLite query parser.
+ * https://github.com/oven-sh/bun/blob/e0aae8adc1ca0d84046f973e563387d0a0abeb4e/src/js/internal/sql/sqlite.ts#L53-L207
```

Comment:
```suggestion
 * A reimplementation of [Bun's SQLite query parser](https://github.com/oven-sh/bun/blob/e0aae8adc1ca0d84046f973e563387d0a0abeb4e/src/js/internal/sql/sqlite.ts#L53-L207).
```

This quite a long link and I'd suggest using a markdown style link here.